hazelcast-client/src/main/java/com/hazelcast/client/util/ClientAddressCancellableDelegatingFuture.java

```diff
@@ -62,18 +62,12 @@ public boolean cancel(boolean mayInterruptIfRunning) {
             Thread.currentThread().interrupt();
         }
 
-        try {
-            if (cancelSuccessful) {
-                setError(new CancellationException());
-                cancelled = true;
-                return true;
-            }
-            return false;
-        } catch (Exception e) {
-            throw rethrow(e);
-        } finally {
-            setDone();
+        if (cancelSuccessful) {
+            complete(new CancellationException());
+            cancelled = true;
```

Comment:
As far as I understand from the Future contract, regardless of result of cancellation, future should be completed with `CancellationException ` and `future.isCancelled()` return `true`.